app/src/main/java/ch/epfl/favo/user/UserUtil.java

```diff
@@ -123,4 +123,20 @@ public DocumentReference getCurrentUserReference(String userId) {
   public void setCollectionWrapper(CollectionWrapper collectionWrapper) {
     collection = collectionWrapper;
   }
+
+  public CompletableFuture<Void> settleTransaction(String accepterId, String requesterId, double reward) {
+    return findUser(accepterId)
+            .thenCompose(
+                    (accepter) -> {
+                      accepter.setBalance(accepter.getBalance() + reward);
+                        findUser(requesterId)
+                              .thenCompose(
+                                      (requester) -> {
+                                        requester.setBalance(requester.getBalance() - reward);
+                                        return updateUser(requester);
+                                      });
+                      return updateUser(accepter);
+                    });
```

Comment:
I'm not sure if this is safe to do for app currency transactions (for example, to prevent double-spending scenarios): are there any guarantees from Firestore on transactions like this?

Reply:
I recommend using this https://firebase.google.com/docs/firestore/manage-data/transactions